The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10824869563/job/30032820347) [(plain)](https://github.com/rust-lang/rust/commit/aff61cdae8ef26941819265ed300611f39f70124/checks/30032820347/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.878 Building wheels for collected packages: reuse
#13 2.879   Building wheel for reuse (pyproject.toml): started
#13 3.134   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.135   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#13 3.135   Stored in directory: /tmp/pip-ephem-wheel-cache-tshbzjeh/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.138 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.551 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.551 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.119 Collecting virtualenv
#13 4.119 Collecting virtualenv
#13 4.163   Downloading virtualenv-20.26.4-py3-none-any.whl (6.0 MB)
#13 4.242      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.0/6.0 MB 78.6 MB/s eta 0:00:00
#13 4.303 Collecting platformdirs<5,>=3.9.1
#13 4.312   Downloading platformdirs-4.3.2-py3-none-any.whl (18 kB)
#13 4.336 Collecting distlib<1,>=0.3.7
#13 4.347      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 86.9 MB/s eta 0:00:00
#13 4.395 Collecting filelock<4,>=3.12.2
#13 4.398   Downloading filelock-3.16.0-py3-none-any.whl (16 kB)
#13 4.398   Downloading filelock-3.16.0-py3-none-any.whl (16 kB)
#13 4.482 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.680 Successfully installed distlib-0.3.8 filelock-3.16.0 platformdirs-4.3.2 virtualenv-20.26.4
#13 DONE 4.8s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      202688 kB
DirectMap2M:     7137280 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/6c65d4f47f82836f303026ec70f752e30d586bd4/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-6c65d4f47f82836f303026ec70f752e30d586bd4-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 29.92s
##[endgroup]
fmt check
Diff in /checkout/src/tools/opt-dist/src/tests.rs:103:
     for test_path in env.skipped_tests() {
         args.extend(["--skip", test_path]);
-    cmd(&args)
-    cmd(&args)
-        .env("COMPILETEST_FORCE_STAGE0", "1")
-        .run()
-        .context("Cannot execute tests")
+    cmd(&args).env("COMPILETEST_FORCE_STAGE0", "1").run().context("Cannot execute tests")
 
 
 /// Tries to find the version of the dist artifacts (either nightly, beta, or 1.XY.Z).
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/src/tools/collect-license-metadata/src/licenses.rs" "/checkout/src/tools/collect-license-metadata/src/reuse.rs" "/checkout/src/tools/collect-license-metadata/src/path_tree.rs" "/checkout/src/tools/collect-license-metadata/src/main.rs" "/checkout/src/tools/run-make-support/src/env.rs" "/checkout/src/tools/run-make-support/src/symbols.rs" "/checkout/src/tools/run-make-support/src/scoped_run.rs" "/checkout/src/tools/run-make-support/src/targets.rs" "/checkout/src/tools/run-make-support/src/macros.rs" "/checkout/compiler/rustc_transmute/src/layout/tree/tests.rs" "/checkout/compiler/rustc_transmute/src/layout/tree.rs" "/checkout/compiler/rustc_transmute/src/layout/dfa.rs" "/checkout/compiler/rustc_transmute/src/layout/mod.rs" "/checkout/compiler/rustc_transmute/src/layout/nfa.rs" "/checkout/compiler/rustc_transmute/src/lib.rs" "/checkout/src/tools/run-make-support/src/external_deps/c_build.rs" "/checkout/src/tools/run-make-support/src/external_deps/rustc.rs" "/checkout/src/tools/run-make-support/src/external_deps/htmldocck.rs" "/checkout/src/tools/run-make-support/src/external_deps/llvm.rs" "/checkout/src/tools/run-make-support/src/external_deps/cc.rs" "/checkout/src/tools/run-make-support/src/external_deps/python.rs" "/checkout/src/tools/run-make-support/src/external_deps/cargo.rs" "/checkout/src/tools/run-make-support/src/external_deps/mod.rs" "/checkout/src/tools/run-make-support/src/external_deps/clang.rs" "/checkout/src/tools/run-make-support/src/external_deps/rustdoc.rs" "/checkout/src/tools/run-make-support/src/assertion_helpers.rs" "/checkout/src/tools/run-make-support/src/string.rs" "/checkout/src/tools/run-make-support/src/lib.rs" "/checkout/src/tools/run-make-support/src/util.rs" "/checkout/src/tools/run-make-support/src/run.rs" "/checkout/compiler/rustc_transmute/src/maybe_transmutable/tests.rs" "/checkout/compiler/rustc_transmute/src/maybe_transmutable/mod.rs" "/checkout/compiler/rustc_transmute/src/maybe_transmutable/query_context.rs" "/checkout/src/tools/run-make-support/src/diff/tests.rs" "/checkout/src/tools/run-make-support/src/diff/mod.rs" "/checkout/src/tools/run-make-support/src/path_helpers.rs" "/checkout/src/tools/run-make-support/src/command.rs" "/checkout/src/tools/run-make-support/src/artifact_names.rs" "/checkout/src/tools/run-make-support/src/fs.rs" "/checkout/src/tools/rls/src/main.rs" "/checkout/src/tools/tier-check/src/main.rs" "/checkout/compiler/rustc_ast/src/ptr.rs" "/checkout/compiler/rustc_ast/src/tokenstream.rs" "/checkout/compiler/rustc_ast/src/entry.rs" "/checkout/compiler/rustc_ast/src/mut_visit.rs" "/checkout/compiler/rustc_ast/src/node_id.rs" "/checkout/compiler/rustc_ast/src/ast_traits.rs" "/checkout/compiler/rustc_ast/src/visit.rs" "/checkout/compiler/rustc_ast/src/token.rs" "/checkout/compiler/rustc_ast/src/lib.rs" "/checkout/compiler/rustc_ast/src/format.rs" "/checkout/compiler/rustc_ast/src/expand/allocator.rs" "/checkout/compiler/rustc_ast/src/expand/mod.rs" "/checkout/src/tools/opt-dist/src/exec.rs" "/checkout/src/tools/opt-dist/src/tests.rs" "/checkout/src/tools/opt-dist/src/training.rs" "/checkout/compiler/rustc_ast/src/attr/mod.rs" "/checkout/src/tools/opt-dist/src/utils/artifact_size.rs" "/checkout/src/tools/opt-dist/src/utils/mod.rs" "/checkout/src/tools/opt-dist/src/utils/io.rs" "/checkout/src/tools/opt-dist/src/metrics.rs" "/checkout/src/tools/opt-dist/src/bolt.rs" "/checkout/src/tools/opt-dist/src/timer.rs" "/checkout/src/tools/bump-stage0/src/main.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Thu Sep 12 05:46:47 UTC 2024
  network time: Thu, 12 Sep 2024 05:46:47 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/bootstrap/src/core/sanity.rs
+++ b/src/bootstrap/src/core/sanity.rs
@@ -233,7 +233,8 @@ than building it.
         }
 
         // Ignore fake targets that are only used for unit tests in bootstrap.
-        if cfg!(not(feature = "bootstrap-self-test")) && !skip_target_sanity {
+        if cfg!(not(feature = "bootstrap-self-test")) && !skip_target_sanity && !build.local_rebuild
+        {
             let mut has_target = false;
             let target_str = target.to_string();
 

--- a/src/tools/opt-dist/src/tests.rs
+++ b/src/tools/opt-dist/src/tests.rs
@@ -67,6 +67,7 @@ change-id = 115898
 [build]
 rustc = "{rustc}"
 cargo = "{cargo}"
+local-rebuild = true
 
 [target.{host_triple}]
 llvm-config = "{llvm_config}"
@@ -102,13 +103,7 @@ llvm-config = "{llvm_config}"
     for test_path in env.skipped_tests() {
         args.extend(["--skip", test_path]);
     }
-    cmd(&args)
-        .env("COMPILETEST_FORCE_STAGE0", "1")
-        // Above we override the stage 0 compiler with previously compiled compiler,
-        // which can cause confusion in bootstrap's target sanity checks.
-        .env("BOOTSTRAP_SKIP_TARGET_SANITY", "1")
-        .run()
-        .context("Cannot execute tests")
+    cmd(&args).env("COMPILETEST_FORCE_STAGE0", "1").run().context("Cannot execute tests")
 }
 
 /// Tries to find the version of the dist artifacts (either nightly, beta, or 1.XY.Z).

